reconcile/change_owners/diff.py

```diff
@@ -252,17 +254,16 @@ def deepdiff_path_to_jsonpath(deep_diff_path: str) -> jsonpath_ng.JSONPath:
     if not deep_diff_path.startswith("root"):
         raise ValueError("a deepdiff path must start with 'root'")
 
-    def build_jsonpath_part(element: str) -> jsonpath_ng.JSONPath:
-        if element.isdigit():
-            return jsonpath_ng.Index(int(element))
+    def build_jsonpath_part(element: Union[str, int]) -> jsonpath_ng.JSONPath:
+        match element:
+            case int():
+                return jsonpath_ng.Index(element)
+            case str():
+                if "." in element:
+                    return jsonpath_ng.Fields(f"'{element}'")
+                return jsonpath_ng.Fields(element)
 
-        if "." in element:
-            return jsonpath_ng.Fields(f"'{element}'")
-        return jsonpath_ng.Fields(element)
-
-    path_parts = [
-        build_jsonpath_part(p) for p in DEEP_DIFF_RE.findall(deep_diff_path[4:])
-    ]
+    path_parts = [build_jsonpath_part(p) for p in parse_path(deep_diff_path)]
```

Comment:
Is `DEEP_DIFF_RE` still used anywhere else?

Reply:
good point, cleaned up